File: GFG/Solution/Medium/42.cs
class Solution {
    public int longestKSubstr(string s, int k) {
        // code here
        if(string.IsNullOrEmpty(s) || k == 0 || k > s.Length)
          return -1;
          
        Dictionary<char, int> charCount = new Dictionary<char,int>();
        int left = 0;
        int maxLength = -1;
        
        for(int right = 0; right < s.Length; right++){
            char rightChar = s[right];
            if(charCount.ContainsKey(rightChar))
               charCount[rightChar]++;
            else
               charCount[rightChar] = 1;
            
            while(charCount.Count > k){
                char leftChar = s[left];
                charCount[leftChar]--;
                if(charCount[leftChar] == 0)
                  charCount.Remove(leftChar);
                left++;
            }
            
            if(charCount.Count == k)
              maxLength = Math.Max(maxLength, right - left + 1);
        }
        return maxLength;
    }
}

File: GFG/Solution/Easy/42.cpp
class Solution {
  public:
    int sumDiffPairs(vector<int>& arr, int k) {
        // code here
        sort(arr.begin(), arr.end());
        int n = arr.size();
        vector<int> dp(n + 1, 0);
        
        for(int i = n - 2; i >= 0; i--){
            dp[i] = dp[i + 1];
            if(arr[i + 1] - arr[i] < k){
                dp[i] = max(dp[i], arr[i] + arr[i + 1] + dp[i + 2]);
            }
        }
        return dp[0];
    }
};